core/src/main/java/tc/oc/pgm/api/Config.java

```diff
@@ -29,11 +30,19 @@
   String getDatabaseUri();
 
   /**
-   * Gets a list of map sources, typically directory paths.
+   * Gets a list of directory paths to locally stored map sources.
    *
-   * @return A list of map sources.
+   * @return A list of folder map sources.
    */
-  List<String> getMapSources();
+  List<String> getFolderMapSources();
+
+  /**
+   * Gets a list of @link{GitRepository} which are used as remote map sources
+   *
+   * @see GitRepository
+   * @return A list of git map sources
+   */
+  List<GitRepository> getGitMapSources();
```

Comment:
I'd change this to be `List<? extends MapSourceFactory> getMapSources()`

Reply:
And only have 1 method.

Reply:
That makes no sense, do you want to move the parsing of `MapSourceFactory`s into `PGMConfig`?

Reply:
Because neither `GitRepository` or `String` extend `MapSourceFactory` :p

Reply:
Yea, move parsing into `PGMConfig`

Reply:
Problem with this: since you said not to include any logging in the Config it removes to the ability to display any helpful messages on any errors with the map sources. ( I can only print the stack traces, which i dont think is very helpful for a normal user).

Reply:
You can throw a `TextException`, with a useful message.